### DISCUSSION TITLE
groupby not working in hunt or alerts

### DISCUSSION THREAD

**Discussion by B3DTech** [Security-Onion-Solutions/securityonion#2546] [Unsupported Versions]
Seems that groupby is not working properly for me in 2.3.21 for either the Hunt dashboard or the alerts dashboard.

<img width="1359" alt="image" src="https://user-images.githubusercontent.com/32655512/103801497-cdadf180-501b-11eb-81d1-1816eec3d1ee.png">

<img width="1359" alt="image" src="https://user-images.githubusercontent.com/32655512/103801612-f930dc00-501b-11eb-9ec4-1b34fe7a6697.png">

**Comment by jertel** [comment]
I noticed the top and "Most Occurrences" and "Fewest Occurrences" graphs are also blank. It looks like Elastic is not providing any of the requested aggregations. I've not seen this behavior so I can't offer an immediate solution or work-around. I suggest reviewing the index data types for source.ip and see if something has changed. A quick test to see if that field has indeed changed would be to change your query to use a `groupby source.ip.keyword` and see if the aggregations show up.

**Reply by B3DTech** [reply]
Strange - not sure what it was like before, but source and destination IP's have a field format of URL?  Changing the query to groupby source.ip.keyword didn't help and it seems to work fine in Kibana directly.  It's just in SOC that I have the issue.

<img width="1016" alt="image" src="https://user-images.githubusercontent.com/32655512/104040812-214c4680-51a6-11eb-9bc8-153dc8fc1a6b.png">

**Reply by jertel** [reply]
Those mappings look ok. 

You can enable debug logging in soc by adding a line to the `/opt/so/conf/soc/soc.json` file:

```
"logLevel": "debug"
```
Ex:
```
{
    "logLevel": "debug",
    "logFilename": "logs/soc.log",
   ...
```
Restarting soc will then show all the request and responses from Elastic. It might help you pinpoint the problem. Be aware though that this will output a lot of information so I suggest first turning the Event and Group Fetch limits down to the lowest setting in the Hunt UI before submitting any queries.
